apps/src/EditorAnnotator.js

```diff
@@ -830,7 +830,7 @@ export default class EditorAnnotator {
       return EditorAnnotator.code_;
     }
 
-    let code = EditorAnnotator.annotator()?.getCode();
+    let code = EditorAnnotator.annotator()?.getCode() || '';
```

Comment:
We can do `getCode(options)?.split('\n')` on line 855 and 860 and then bail out of `findCodeRegion`

(line 891)
```
    let code = EditorAnnotator.getCode(options);
    let lines = EditorAnnotator.getLines(options);

    if (!code && !lines) {
      return ret;
    }

    // Attempt to just find the code in the full code listing
```

Reply:
It won't "recover" and parse the actual code that's loaded until you the first time you cycle through a learning goal with evidence. (but allowing `getCode` to return an empty string allows `lines` to be cached with that empty string, causing it to never parse evidence, I believe)

Reply:
I'm happy to try this... however one thing to point out is that `''` is falsy in js, so I'm not sure I see where `lines` would be cached as `['']`? knowing that... lmk if you still have concerns with the fix as I have it in this PR.

Reply:
Sorry. Didn't see this.

Yes. `''` is falsy. However, `getLines` will take that `''` returned from `getCode` and split it.

`''.split('\n')` will become `['']` which, in JS, is truthy.

The code will cache that result in `getLines` and it will go out of sync with the non-cached result of `getCode`.

Reply:
If we keep your fix as is... the page will load but never recover enough to show any evidence because `getLines` will cache the array with that empty string. We have to make sure neither cache it.